src/Writer/Mongo.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Log\Writer;
+
+use DateTimeInterface;
+use Mongo as MongoC;
+use MongoClient;
+use MongoCollection;
+use MongoDate;
+use Traversable;
+use Zend\Log\Exception;
+use Zend\Log\Formatter\FormatterInterface;
+use Zend\Stdlib\ArrayUtils;
+
+/**
+ * Mongo log writer.
+ */
+class Mongo extends AbstractWriter
+{
+    /**
+     * MongoCollection instance
+     *
+     * @var MongoCollection
+     */
+    protected $mongoCollection;
+
+    /**
+     * Options used for MongoCollection::save()
+     *
+     * @var array
+     */
+    protected $saveOptions;
+
+    /**
+     * Constructor
+     *
+     * @param MongoC|MongoClient|array|Traversable $mongo
+     * @param string $database
+     * @param string $collection
+     * @param array $saveOptions
+     * @throws Exception\InvalidArgumentException
+     * @throws Exception\ExtensionNotLoadedException
+     */
+    public function __construct($mongo, $database = null, $collection = null, array $saveOptions = [])
+    {
+        if (!extension_loaded('mongo')) {
+            throw new Exception\ExtensionNotLoadedException('Missing ext/mongo');
+        }
+
+        if ($mongo instanceof Traversable) {
+            // Configuration may be multi-dimensional due to save options
+            $mongo = ArrayUtils::iteratorToArray($mongo);
+        }
+        if (is_array($mongo)) {
+            parent::__construct($mongo);
+            $saveOptions = isset($mongo['save_options']) ? $mongo['save_options'] : [];
+            $collection  = isset($mongo['collection']) ? $mongo['collection'] : null;
+            $database    = isset($mongo['database']) ? $mongo['database'] : null;
+            $mongo       = isset($mongo['mongo']) ? $mongo['mongo'] : null;
```

Comment:
Wish we were on PHP 7 as the minimum version, so we could use `??` here. :frowning: